HashRegistrarSimplified.sol

```diff
@@ -460,6 +464,43 @@ contract Registrar {
     }
 
     /**
+     * @dev Allows anyone to delete the owner and resolver records for a (subdomain of) a
+     *      name that is not currently owned in the registrar. If passing, eg, 'foo.bar.eth',
+     *      the owner and resolver fields on 'foo.bar.eth' and 'bar.eth' will all be cleared.
+     * @param labels A series of label hashes identifying the name to zero out, rooted at the
+     *        registrar's root. Must contain at least one element. For instance, to zero 
+     *        'foo.bar.eth' on a registrar that owns '.eth', pass an array containing
+     *        [sha3('foo'), sha3('bar')].
+     */
+    function eraseNode(bytes32[] labels) {
+        if(labels.length == 0) throw;
+        if(state(labels[labels.length - 1]) == Mode.Owned) throw;
+
+        _eraseNodeHierarchy(labels.length - 1, labels, rootNode);
+    }
+
+    function _eraseSingleNode(bytes32 label) internal {
+        ens.setSubnodeOwner(rootNode, label, address(this));
+        var node = sha3(rootNode, label);
+        ens.setResolver(node, 0);
+        ens.setOwner(node, 0);
+    }
+
+    function _eraseNodeHierarchy(uint idx, bytes32[] labels, bytes32 node) internal {
+        // Take ownership of the node
+        ens.setSubnodeOwner(node, labels[idx], address(this));
+        node = sha3(node, labels[idx]);
+        
+        // Recurse if there's more labels
+        if(idx > 0)
+            _eraseNodeHierarchy(idx - 1, labels, node);
```

Comment:
I'd be interested to know, at 4.7M gas, how 'deep' can this function reach, at a maximum? I wonder if it's in the tens or the hundreds.. The SSTORE's must be quite costly, more so than the SHAs. My back-of-the-envelope calculation suggests ~100 levels.